stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -64,12 +67,38 @@ func posixPipe() -> (readFD: CInt, writeFD: CInt) {
 /// stderr.
 public func spawnChild(args: [String])
   -> (pid: pid_t, stdinFD: CInt, stdoutFD: CInt, stderrFD: CInt) {
+  let childStdout = posixPipe()
+  let childStdin = posixPipe()
+  let childStderr = posixPipe()
+
+#if os(Android)
+  // posix_spawn isn't available on Android. Instead, we fork and exec.
+  let pid = fork()
+  if pid < 0 {
+    preconditionFailure("fork() failed")
+  } else if pid == 0 {
+    // pid of 0 means we are now in the child process.
+    // Capture the output and execute the program.
+    dup2(childStdout.writeFD, STDOUT_FILENO)
+    dup2(childStdin.readFD, STDIN_FILENO)
+    dup2(childStderr.writeFD, STDERR_FILENO)
+
+    var result: Int32 = 0
+    withArrayOfCStrings([Process.arguments[0]] + args) {
+      result = execve(Process.arguments[0], $0, _getEnviron())
+    }
+
+    if result != 0 {
+      preconditionFailure("execve() failed. errno: \(errno)")
+    }
+    _exit(126)
```

Comment:
Finally getting around to this, sorry for the wait. @weissi, could you explain `child2parent` in greater detail? Would the child process file descriptors be duped to write to the parent pipe in some way? I think I'm missing something.